index.js

```diff
@@ -256,6 +258,9 @@ const ContainerService = {
 
     webExtension.startup().then(api => {
       api.browser.runtime.onMessage.addListener((message, sender, sendReply) => {
+        if (message.method === "checkForMajorUpgrade") {
+          sendReply(reason === "upgrade" && MAJOR_VERSIONS.indexOf(self.version) > -1);
```

Comment:
The following will work in the background code:
```
const MAJOR_VERSIONS = ["2.2.0"];

async function checkExtensionInfo() {
  const manifestPath = browser.extension.getURL("manifest.json");
  const response = await fetch(manifestPath);
  const extensionInfo = response.json();
  if (MAJOR_VERSIONS.indexOf(extensionInfo.version) > -1)) {
    browser.browserAction.setBadgeBackgroundColor({color: "rgba(0,217,0,255)"});
    browser.browserAction.setBadgeText({text: "NEW"});
  }
}
checkExtensionInfo()
```

This would need some idea of updating though... perhaps a similar local storage to clear when the user has opened the browser action.